### ISSUE TITLE
SecureStringArgument throws exception when Console.In is redirected

### ISSUE THREAD

**Issue by mbettex** [adamabdelhamed/PowerArgs#97]
When Console.In is redirected and you use SecureStringArgument, an exception is thrown.

For instance, in my application I have the following arguments defined:

```
public class Arguments
{
    [ArgDescription("The username")]
    [ArgShortcut("u")]
    public string Username { get; set; }

    [ArgDescription("The password.")]
    [ArgShortcut("p")]
    public SecureStringArgument Password { get; set; }
}
```

And my main is similar to this

```
private static int Main(string[] args)
{
    var arguments = Args.Parse<Arguments>(args);

    var username = arguments.Username;
    var password = arguments.Password.ConvertToNonsecureString();

    // More stuff...
}
```

I run the application from another process  using the following code:

```
var arguments = "-u myusername";
var stdIn = "mypassword";

string stdOut;
string stdErr;

using (var process = new Process())
{
    process.StartInfo.FileName = "myPowerArgsApp.exe";
    process.StartInfo.Arguments = arguments;
    process.StartInfo.RedirectStandardInput = true;
    process.StartInfo.RedirectStandardOutput = true;
    process.StartInfo.RedirectStandardError = true;
    process.StartInfo.UseShellExecute = false;
    process.StartInfo.CreateNoWindow = true;

    process.Start();

    process.StandardInput.WriteLine(stdIn);

    using (var processWaiter = Task.Run(() => process.WaitForExit((int)TimeSpan.FromSeconds(10).TotalMilliseconds)))
    using (var outputReader = Task.Run(() => process.StandardOutput.ReadToEnd()))
    using (var errorReader = Task.Run(() => process.StandardError.ReadToEnd()))
    {
        var waitResult = processWaiter.Result;
        if (!waitResult)
        {
            process.Kill();
        }

        stdOut = outputReader.Result;
        stdErr = errorReader.Result;

        if (!waitResult)
        {
            throw new TimeoutException("Process wait timeout expired");
        }
    }
}
```

And the line that reads the secure string throws an `System.InvalidOperationException` with the message 

> Cannot read keys when either application does not have a console or when console input has been redirected from a file. Try Console.Read.

It is because `Console.ReadKey()` does not work when stdin is redirected. That behavior is documented in msdn [here](https://msdn.microsoft.com/en-us/library/471w8d85%28v=vs.110%29.aspx).

Right now, I use the following workaround to have it working in my application:

```
private static int Main(string[] args)
{
    var arguments = Args.Parse<Arguments>(args);

    var username = arguments.Username;

    string password;
    if (Console.IsInputRedirected)
    {
        Console.Write("Enter value for Password: ");
        password = Console.ReadLine();
    }
    else
    {
        password = arguments.Password.ConvertToNonsecureString();
    }

    // More stuff...
}
```

I believe that having this working out of the box would be really great in order to make it possible to launch applications that uses PowerArgs and SecureStringArgument from other applications. Maybe something similar to the workaround I have implemented in my own code could be included directly in PowerArgs?

**Comment by adamabdelhamed** [comment]
The only problem with that workaround is that it will result in the password being displayed in the console.  That sort of defeats the purpose of using a SecureString, right?

**Comment by adamabdelhamed** [comment]
I'm not going to make any change here.  SecureString arguments are for interactive programs where you don't want the password to be shown.  If your program can get the string from either an interactive session or via redirected input then those should be two different branches in your program.

**Comment by mbettex** [comment]
Sorry, I went to holidays after opening the issue and forgot to follow up when I came back.

In my case here, I have an application that requires a password to connect to some third party service. Sometimes, it is used by a real user and in that case, the current SecureString behavior is perfect, as the password can never be displayed in the console.

The same application is also launched by another application, and the input and output stream are redirected from and to this parent application. In that case, there is no console where the password might be shown. That's the case that is throwing the exception.

It would have been easier for me to have both scenarios working out of the box. But anyway, I'm using the workaround that I described above which works fine for me, so this is not a blocker.

Thank you for your time and this otherwise excellent library :-)
